components/data-services/org.wso2.carbon.dataservices.capp.deployer/src/main/java/org/wso2/carbon/dataservices/capp/deployer/internal/DataServiceCappDeployerServiceComponent.java

```diff
@@ -49,19 +54,32 @@ protected synchronized void deactivate(ComponentContext ctx) {
         }
     }
 
-    protected void setDataSrcCappDeployer(AppDeploymentHandler appDeploymentHandler) {
+    protected void setDataServiceCappDeployer(AppDeploymentHandler appDeploymentHandler) {
         if (appDeploymentHandler instanceof DataSourceCappDeployer) {
-            try {
-                //register data source deployer as an OSGi service
-                DataServiceCappDeployer dataServiceCappDeployer = new DataServiceCappDeployer();
-                this.ctx.getBundleContext().registerService(AppDeploymentHandler.class.getName(),
-                                                            dataServiceCappDeployer, null);
-            } catch (Throwable e) {
-                log.error("Failed to activate Data Service Capp Deployer", e);
+            if (Objects.isNull(ctx)) {
+                // save appDeploymentHandler
+                appDepHandler = appDeploymentHandler;
+            } else {
+                registerDataServiceCappDeployer();
             }
         }
     }
 
-    protected void unsetDataSrcCappDeployer(AppDeploymentHandler appDeploymentHandler) {
+    protected void unsetDataServiceCappDeployer(AppDeploymentHandler appDeploymentHandler) {
+        if (appDeploymentHandler.equals(appDepHandler)) {
+            appDepHandler = null;
+        }
+    }
+
+    /**
+     * register data source deployer as an OSGi service
```

Comment:
First line of Doc comments need to start with an uppercase letter and end with a period. Shall we correct in all places? Please refer to https://www.oracle.com/technetwork/java/codeconventions-150003.pdf for code conventions

Reply:
Fixed in https://github.com/wso2/carbon-data/pull/254